test/Elastic.Apm.Tests/StackTraceTests.cs

```diff
@@ -175,18 +176,260 @@ public void InheritedChain()
+				payloadSender.FirstSpan.StackTrace[0].Function.Should().Be("End");
+				payloadSender.FirstSpan.StackTrace[1].Function.Should().Be("CaptureSpan");
+			});
+
+		[Fact]
+		public void StackTraceLimit0SpanFramesMinDuration0() =>
+			AssertWithAgent("0", "0", payloadSender =>
+			{
+				payloadSender.FirstSpan.Should().NotBeNull();
+				payloadSender.FirstSpan.StackTrace.Should().BeNull();
+			});
+
+		[Fact]
+		public void StackTraceLimitNegativeSpanFramesMinDuration0() =>
+			AssertWithAgent("-1", "0", payloadSender =>
+			{
+				payloadSender.FirstSpan.Should().NotBeNull();
+				payloadSender.FirstSpan.StackTrace.Should().BeNull();
+			});
+
+		[Fact]
+		public void StackTraceLimit2SpanFramesMinDuration0() =>
+			AssertWithAgent("2", "0", payloadSender =>
+			{
+				payloadSender.FirstSpan.Should().NotBeNull();
+				payloadSender.FirstSpan.StackTrace.Should().BeNull();
+			});
+
+		[Fact]
+		public void StackTraceLimit0SpanFramesMinDuration100() =>
+			AssertWithAgent("0", "100", payloadSender =>
+			{
+				payloadSender.FirstSpan.Should().NotBeNull();
+				payloadSender.FirstSpan.StackTrace.Should().BeNullOrEmpty();
+			}, 150);
+
+		[Fact]
+		public void StackTraceLimitNegativeSpanFramesMinDuration100() =>
+			AssertWithAgent("-1", "100", payloadSender =>
+			{
+				payloadSender.FirstSpan.Should().NotBeNull();
+				payloadSender.FirstSpan.StackTrace.Should().NotBeEmpty();
+				payloadSender.FirstSpan.StackTrace.Should().HaveCountGreaterThan(10);
+
+				//We don't filter out frames from the agent, therefore the current implementation will contain 2 agent frames on top
+				payloadSender.FirstSpan.StackTrace[0].Function.Should().Be("End");
+				payloadSender.FirstSpan.StackTrace[1].Function.Should().Be("CaptureSpan");
+			}, 150);
+
+		[Fact]
+		public void StackTraceLimit2SpanFramesMinDuration100() =>
+			AssertWithAgent("2", "100", payloadSender =>
+			{
+				payloadSender.FirstSpan.Should().NotBeNull();
+				payloadSender.FirstSpan.StackTrace.Should().NotBeEmpty();
+				payloadSender.FirstSpan.StackTrace.Should().HaveCount(2);
+
+				//We don't filter out frames from the agent, therefore the current implementation will contain 2 agent frames on top
+				payloadSender.FirstSpan.StackTrace[0].Function.Should().Be("End");
+				payloadSender.FirstSpan.StackTrace[1].Function.Should().Be("CaptureSpan");
+			}, 150);
+
+		[Fact]
+		public void DefaultStackTraceLimitAndSpanFramesMinDuration_ShortSpan()
+		{
+			var payloadSender = new MockPayloadSender();
+
+			using (var agent =
+				new ApmAgent(new TestAgentComponents(
+					new TestAgentConfigurationReader(new NoopLogger()), payloadSender)))
+			{
+				agent.Tracer.CaptureTransaction("TestTransaction", "Test", t
+					=>
+				{
+					t.CaptureSpan("span", "span", () => { });
+				});
+			}
+
+			payloadSender.FirstSpan.Should().NotBeNull();
+			payloadSender.FirstSpan.StackTrace.Should().BeNullOrEmpty();
```

Comment:
Because this tests the default config values and the assumption is that the empty span will be less than 5ms, therefore no stacktrace will be captured. (more in the next comment :) )

Reply:
What I meant is that if stack trace is not captured then it will be null but how it can be empty? So shouldn't the line
```
payloadSender.FirstSpan.StackTrace.Should().BeNullOrEmpty();
```
be 
```
payloadSender.FirstSpan.StackTrace.Should().BeNull();
```
?

Reply:
Och, I see. 

Yes, that's correct, but I thought it's more like an implementation detail - it doesn't really matter if it's an empty `List` or `null` - the point is that there are no stack frames. Seems more robust to me.

Reply:
But stack trace cannot be empty by definition - stack trace for a span contains at least the application entry method (`Program.Main`) and stack trace for an error contains at least the method where exception is caught.

Reply:
Yes, but the meaning of "no stack trace captured" can be represented as a list containing no item. - sometimes people want to avoid `null`. We don't currently do that, and I think that's good, but that's just an implementation detail.

But this is really a minor detail - if you want I can change it. I personally think it's more robust since I don't wanted to test on `null`, I wanted to test on not having stack frames, but I'm fine with changing it.

Reply:
Not a major point - up to you. I understand that empty container is sometimes better than `null` as Null Object pattern application (like for example empty C# `event` without handlers) but I think in this case `null` is clearer.

In any case it might be worth making sure it's consistent with the rest of of the test cases.